Automattic/jetpack
# Jetpack Boost: DataSync Error Handling - Phase 2

## Proposed changes:

Followup to #35062

This refactors how fallback conditions are handled and introduces `Schema_Context` to allow logging information while parsing and then delivering that information along with the parsed results. This allows for more graceful error handling and easier debugging.

Constants introduced:
```
// This enables DataSync logger
define( 'DATASYNC_DEBUG', true );
// This makes the logger very noisy, but might be useful for debugging
define( 'DATASYNC_VERBOSE', true );
// This enables debug_trace on datasync logs
define( 'DATASYNC_TRACE', 20 );
```


#### 🦋  Stack Trace
@thingalon introduced me to his tiny stack trace helper function. Does it make sense to keep it here hidden behind a `DATASYNC_TRACE` flag?


### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:

1. Apply patch
2. Run unit tests
3. Enable the constants above one by one - test out the Jetpack Boost plugin, make sure everything works.
